maxAreaOfIsland.js

```diff
@@ -0,0 +1,126 @@
+/*
+You are given an m x n binary matrix grid. An island is a group of 1's (representing land) connected 4-directionally (horizontal or vertical.) 
+You may assume all four edges of the grid are surrounded by water.
+
+The area of an island is the number of cells with a value 1 in the island.
+
+Return the maximum area of an island in grid. If there is no island, return 0.
+*/
+
+//    0  1  2  3  4  5  6  7  8  9  10 11 12
+// 0 [0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+// 1 [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
+// 2 [0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+// 3 [0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0],
+// 4 [0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0],
+// 5 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+// 6 [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
+// 7 [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0],
+
+const { Queue } = require('./utils/queue');
+
+const directions = [
+    [-1, 0],
+    [0, -1],
+    [0, 1],
+    [1, 0],
+];
+
+const ISLAND = 1;
+const WATER = 0;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isWithinBounds = (newRow, newCol, grid) => {
```

Comment:
There's no reason to name this `newRow`. We want this function to be generic. Anyone can pass in a row, col, and a grid and we will tell them if that row, col pair is in bounds. A user of this function does not know anything about what a `newRow` is, and they might get confused when reading this code.